### ISSUE TITLE
No way of configuring Scope in connections.py

### ISSUE THREAD

**Issue by olaor** [O365/python-o365#80]
Connections.oauth2 has no way of changing the scope for the token, as it is hardcoded.

I wanted to read my Calendars, but since Calendars.Read was not requested I was not allowed even if I gave the app those permissions. So I had to hardcode that in the source as well.

**Comment by olaor** [comment]
Err. I meant connection.py and Connection.oauth2. My bad.

**Comment by Narcolapser** [comment]
Can you post some code of what you are doing?

**Comment by olaor** [comment]
Yes, sorry for being so slow. 

The current connection.py looks like this:
```
 connection.oauth = OAuth2Session(client_id=client_id,
                                             redirect_uri='https://outlook.office365.com/owa/',
                                             scope=['https://graph.microsoft.com/Mail.ReadWrite',
                                                    'https://graph.microsoft.com/Mail.Send',
'offline_access'], )
```
When trying to read Calendars one needs the permission Calendars.read, and you have to ask for that specifically when creating a token even if one allows Calendar.Read in O365 App settings. So I manually added this:

```
connection.oauth = OAuth2Session(client_id=client_id,
                                             redirect_uri='https://outlook.office365.com/owa/',
                                             scope=['https://graph.microsoft.com/Mail.ReadWrite',
                                                    'https://graph.microsoft.com/Mail.Send',
                                                    'https://graph.microsoft.com/Calendars.Read',
                                                    'offline_access'], )
```

There should be a way to pass a list of scopes to the `Connection.oauth2()` function. Having to hardcode this in the source code breaks it's maintainability :)

Great lib by the way!

**Comment by roycem90** [comment]
This is available in v1.0
